plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/AltibaseTable.java

```diff
@@ -0,0 +1,282 @@
+                } else {
+                    query.append(tableAliasName).append(col.getName()).append(" as ").append(col.getName());
+                }
+            }
+        } catch (DBException e) {
+            log.warn(e);
+        }
+    }
+
+    @Property(viewable = false, hidden = true)
+    public String getTableType() {
+        return super.getTableType();
+    }
+    
+    @Property(viewable = true, order = 20, editable = false, formatter = ByteNumberFormat.class, category = DBConstants.CAT_STATISTICS)
+    public Long getTableSize(DBRProgressMonitor monitor) throws DBCException {
+        if (!hasStatistics()) {
+            loadSize(monitor);
+        }
+
+        return tableSize[SIZE_IDX_MEM] + tableSize[SIZE_IDX_DISK];
+    }
+    
+    @Property(viewable = true, order = 22, editable = false, formatter = ByteNumberFormat.class)
+    public Long getTableSizeInMemory(DBRProgressMonitor monitor) throws DBCException {
+        if (!hasStatistics()) {
+            loadSize(monitor);
+        }
+
+        return tableSize[SIZE_IDX_MEM];
+    }
+    
+    @Property(viewable = true, order = 23, editable = false, formatter = ByteNumberFormat.class)
+    public Long getTableSizeInDisk(DBRProgressMonitor monitor) throws DBCException {
+        if (!hasStatistics()) {
+            loadSize(monitor);
+        }
+
+        return tableSize[SIZE_IDX_DISK];
+    }
+    
+    @Override
+    public DBSObject refreshObject(@NotNull DBRProgressMonitor monitor) throws DBException {
+        tableSize = null;
+        getTableSize(monitor);
+        getTablespace(monitor);
+
+        tablePrivCache.clearCache();
+        
+        return super.refreshObject(monitor);
+    }
+    
+    ///////////////////////////////////
+    // Tablespace
+    @Property(viewable = true, order = 15, editable = false)
+    public String getTablespace(DBRProgressMonitor monitor) throws DBException {
+        try (JDBCSession session = DBUtils.openMetaSession(monitor, this, "Load tablespace")) {
+            try (JDBCPreparedStatement dbStat = session.prepareStatement(
+                "SELECT tbs_name, is_partitioned FROM system_.sys_tables_ t, system_.sys_users_ u "
+                + "WHERE u.user_id = t.user_id AND u.user_name = ? AND t.table_name = ?")) {
+                dbStat.setString(1, getSchema().getName());
+                dbStat.setString(2, getName());
+
+                try (JDBCResultSet dbResult = dbStat.executeQuery()) {
+                    if (dbResult.next()) {
+                        tablespace = JDBCUtils.safeGetString(dbResult, 1);
+                        partitioned = JDBCUtils.safeGetBoolean(dbResult, 2, AltibaseConstants.RESULT_Y_VALUE);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            throw new DBCException("Error reading tablespace naame", e);
+        }
+        
+        return tablespace;
```

Comment:
Okay, it would be better to cache it as you recommended. The tablespace of a table can be changed, but in very rare cases, then I can assume that the user refreshes the target table in that case. Thanks for your tip.